### PR TITLE
Update vulnerabilities

### PR DESCRIPTION
**Pull request by obohaciak**
**Related command**
N/A

**Description**
Fixes vulnerabilities in mcr.microsoft.com/azure-cli:2.49.0. Affected packages are:
-   cryptography 38.0.4 (pip)
-   requests 2.26.0 (pip)
-   openssl 3.1.0-r4 (apk)
-   binutils 2.40-r6 (apk)

**Testing Guide**
N/A

**History Notes**
N/A

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by bebound** [comment]
`requests` is bumped in https://github.com/Azure/azure-cli/pull/26571
`openssl` and `binutils` will be updated once there is new version in Alpine repo.

The reason of using old version cryptography is in https://github.com/Azure/azure-cli/pull/25690. We are working on Windows 64-bit MSI package in #26640, once it's finished, we'll bump it.

File: Dockerfile [hunk 1]
```diff
@@ -63,6 +63,12 @@ RUN ./scripts/install_full.sh && python ./scripts/trim_sdk.py \
     )" \
  && apk add --virtual .rundeps $runDeps
 
+# update vulnerabilities
+RUN apk upgrade openssl
+RUN apk upgrade binutils --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
+RUN pip install cryptography -U
+RUN pip install requests -U
```
Comment (jiasli) [inline Dockerfile:70]: This will be covered by https://github.com/Azure/azure-cli/pull/26671.

File: Dockerfile [hunk 2]
```diff
@@ -63,6 +63,12 @@ RUN ./scripts/install_full.sh && python ./scripts/trim_sdk.py \
     )" \
  && apk add --virtual .rundeps $runDeps
 
+# update vulnerabilities
+RUN apk upgrade openssl
+RUN apk upgrade binutils --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
```
Comment (jiasli) [inline Dockerfile:68]: This is not a universal solution as they are only specific to certain versions of Alpine Linux image. Alpine Linux will include these fixes in its new docker image. We also automatically pick up the latest Alpine Linux image: https://github.com/Azure/azure-cli/blob/5f52c83e218f3ddc4d933bd99a88dab9e1864d81/Dockerfile#L8

**Pull request closed without merging**